### ISSUE TITLE
Spacing when nesting directives.

### ISSUE THREAD

**Issue by lezcano** [pytorch/pytorch_sphinx_theme#129]
The spacing when nesting a `.. code` directive inside a `warning` or a `note` is not rendered correctly. 

See for example the end of the first warning in https://pytorch.org/docs/stable/generated/torch.qr.html

![image](https://user-images.githubusercontent.com/3291265/123652280-12288600-d824-11eb-9365-b86e7b2e7ff5.png)

cc: @mattip

**Comment by mattip** [comment]
As far as I can tell, the problem is that either the `div.admonition` class needs a `padding-bottom: 1px;` value to clearly delineate where the code box ends.

**Comment by mattip** [comment]
@brianjo thoughts?

**Comment by mattip** [comment]
#135 did not fix this and probably should be backed out. I am not sure how to get the `padding-bottom` into the `div.admonition`.

Edit #138 was meant to fix this, but I seem to have forgotten that part of #135.

**Comment by mattip** [comment]
This seems to be fixed on [latest master](https://pytorch.org/docs/master/generated/torch.qr.html), although there probably is still too much padding around the `<pre>`.

**Comment by lezcano** [comment]
It still does not render well in my browser:
![image](https://user-images.githubusercontent.com/3291265/130234743-86f0c639-9183-44a5-b395-d84d7da5c477.png)

The difference I see is that now the background is in a slightly darker grey, and the formulae in double backticks do render as if they had both a rectangle around each letter and a large rectangle (of a slightly lighter grey) around the whole formula.

**Comment by mattip** [comment]
The background color is a different issue #130. This one is about the overlap between the bottom-most `<pre>` containing `Q,R = torch.linalg.qr(A, mode="complete")` and the surrounding `WARNING` admonition, which now line up.

**Comment by lezcano** [comment]
Yes, sorry, I wrote that in both the issues. All I meant to write here is that the spacing between the end of `.. warning::` and the end of `.. code::` is still not rendered correctly.

**Comment by mattip** [comment]
What would you consider to be correct? The bottom of the `..code::` directive previously extended past the surrounding `warning::` directive and no longer does.

**Comment by lezcano** [comment]
Oh, In my head correct would be to leave an space the size of that just before `..code::` so that it does not look crammed. this space would have the color of the background of `..warning::`. Sorry for the misunderstanding.

**Comment by mattip** [comment]
Thanks for the clarification. Makes sense.
